### PR TITLE
Added export methods to project class

### PR DESCRIPTION
**Pull request by SlightlyUnorthodox**
So far I've added the remaining export methods and added a **project_info** attribute of the ```Project``` class which stores the results of the **Export Project Info** call as a dictionary upon initialization.

While everything appears to work, I'm still looking into writing some testing, but I figured I'd send this up the line for you to look over and to make sure nothing is too far off base.

Function added:

* Export Project Info
* Export Report
* Export Instruments
* Export PDF file of Instruments
* Export Survey Link
* Export Survey Queue Link
* Export Survey Return Code
* Export Participant List
* Export Field Names

### CODE REVIEW

File: redcap/project.py [hunk 1]
```diff
@@ -149,6 +157,423 @@ def _call_api(self, payload, typpe, **kwargs):
         rcr = RCRequest(self.url, payload, typpe)
         return rcr.execute(**request_kwargs)
 
+    def __toggle_dataframe(self,format):
+        ret_format = format
+        if format == 'df':
+            try:
```
Comment (sburns) [inline redcap/project.py]: This  can be hoisted to the top of the file. If pandas doesn't import, catch and set `read_csv` to None. Then in export methods we can `if not read_csv and format == 'df': blow up`.

File: redcap/project.py [hunk 2]
```diff
@@ -149,6 +157,423 @@ def _call_api(self, payload, typpe, **kwargs):
...
+
+        if format in ('json', 'csv', 'xml'):
+            return response
+        elif format == 'df':
+            if not df_kwargs:
+                return read_csv(StringIO(response))
+            else:
+                return read_csv(StringIO(response), **df_kwargs)   
+
+    def export_report(self, report_id, format='json',raw_or_label='raw',raw_or_label_headers='raw',export_checkbox_labels=False,df_kwargs=None):
```
Comment (sburns) [inline redcap/project.py]: Please include a space after the `,` in lists, function arguments, dictionaries, etc.

File: redcap/project.py (hunk 1 above)
Comment (sburns) [inline redcap/project.py]: This is actually a good place to put that logic assuming all of the export methods will use it.

File: redcap/project.py [hunk 3]
```diff
@@ -149,6 +157,423 @@ def _call_api(self, payload, typpe, **kwargs):
...
+
+        if format in ('json', 'csv', 'xml'):
+            return response
+        elif format == 'df':
+            if not df_kwargs:
+                return read_csv(StringIO(response))
+            else:
+                return read_csv(StringIO(response), **df_kwargs)   
+
+    def export_pdf(self, format='json',record=None,event=None,instrument=None,all_records=None,df_kwargs=None):
```
Comment (sburns) [inline redcap/project.py]: I don't think this needs a format argument right?

**Comment by sburns** [comment]
I can't believe the test projects are still around, I haven't been at Vandy for almost 2 years

File: redcap/project.py (hunk 3 above)
Comment (SlightlyUnorthodox) [inline redcap/project.py]: I'd have to double-check, and I'm having some hiccups rebuilding my vagrant REDCap instance at the moment, but I believe it still accepts a format argument. 

If I recall correctly, the error message will be handled to that format, which might be useful depending on if more elaborate error-handling is added later.

Comment (sburns) [inline redcap/project.py]: Yeah that sounds right.

File: redcap/project.py (hunk 1 above)
Comment (SlightlyUnorthodox) [inline redcap/project.py]: I like that, it seemed rather awkward as it was. Once I've finished adding methods, I'll try and see if there's anything else that can simplified similarly.

**Comment by SlightlyUnorthodox** [comment]
I added at least a semi-function method for each of the remaining REDCap API methods. While they all appear to work, there doesn't seem to be any coherent way to test them until the **Create Project** method is working, which may be awhile due to undocumented bugs.

* Import Arms
* Import Events
* Import Instrument-Event Mappings
* Import Metadata
* Import Users
* Delete Arms
* Delete Events

**Comment by sburns** [comment]
> which may be awhile due to undocumented bugs.

On REDCap's side?

**Comment by SlightlyUnorthodox** [comment]
Yes. For whatever reason, REDCap fails to recognize that the supertoken grants the permissions to use the ```Create Project``` method. Every method I've tried returns the following: ```{u'error': u'You do not have permissions to use the API'```

**Comment by sburns** [comment]
Can a super user token connect to a project normally? Or is that token only to be used to create projects?

**Comment by SlightlyUnorthodox** [comment]
The supertoken is only used to create projects. The ```Create Project``` and ```Export Project Info``` API calls use the same ```content = 'project' ``` parameter, with the length of the token determining which method is used.

**Comment by sburns** [comment]
Well that's one way to do it.

**Comment by sburns** [comment]
Hey Dax, sorry to drop the ball on this? Can you back-merge master?

**Pull request closed without merging**